src/arm-codegen.c

```diff
@@ -443,8 +445,14 @@ void emit_ph2_ir(ph2_ir_t *ph2_ir)
         emit(__mov_i(__EQ, rd, 1));
         return;
     case OP_log_and:
-        /* FIXME: bad logical-and instruction */
-        emit(__and_r(__AL, rd, rn, rm));
+        /* TODO: Short-circuit evaluation */
```

Comment:
You do have the short circuit evaluation here. The previously mentioned issue is caused by parser, not codegen.
You can remove this TODO here.